js-array-methods-2/js-array-methods-2_animal-objects/index.js

```diff
@@ -91,22 +100,79 @@ const indexOfAnimalWithNameLongerFive = null;
 // Hint: sort() mutates the original array, which is bad.
 // -> Use animals.slice().sort(...) to make a copy (and the tests work).
 
-const animalsSortedAlphabetically = null;
+const animalsSortedAlphabetically = animals.slice().sort((a, b) => {
+  const nameA = a.name.toUpperCase(); // ignore upper and lowercase
+  const nameB = b.name.toUpperCase(); // ignore upper and lowercase
+  if (nameA < nameB) {
+    return -1;
+  }
+  if (nameA > nameB) {
+    return 1;
+  }
 
-const animalsSortedByWeightStartingWithLowest = null;
+  // names must be equal
+  return 0;
+});
 
-const animalsSortedByWeightReversed = null;
+console.log("Original array: ", animals);
+console.log(
+  "Sorted alphabetically by animal name: ",
+  animalsSortedAlphabetically
+);
 
-const animalWithWeightMoreThanFivehundredExists = null;
+const animalsSortedByWeightStartingWithLowest = animals
+  .slice()
+  .sort((a, b) => a.weight - b.weight);
+
+console.log(
+  "Sorted by weight ascending: ",
+  animalsSortedByWeightStartingWithLowest
+);
+
+const animalsSortedByWeightReversed = animals
+  .slice()
+  .sort((a, b) => b.weight - a.weight);
+
+console.log("Sorted by weight descending: ", animalsSortedByWeightReversed);
+
+const animalWithWeightMoreThanFivehundredExists = animals.some(
+  (animal) => animal.weight > 500
+);
+console.log(
+  "Animal that weighs more than 500?: ",
+  animalWithWeightMoreThanFivehundredExists
+);
 
 // Hint: Filter for Europe first, then check every animal for its weight.
-const allAnimalsInEuropeWeighLessThanOnehundred = null;
+const allAnimalsInEuropeWeighLessThanOnehundred = animals
```

Comment:
Unfortunately, the variable name `allAnimalsInEuropeWeighLessThanOnehundred` is misleading since the test expects the boolean `false` instead of animals matching the filter conditions. I agree with your solution. But a correct solution conform to the test case would be:

`const allAnimalsInEuropeWeighLessThanOnehundred = animals
  .filter((animal) =>
    animal.continents.find((continent) => continent === "Europe")
  )
  .every((europeanAnimal) => europeanAnimal.weight < 100);`
  
This is returning `false` since NOT ALL animals in Europe are weighing less than one hundred units, which is the actual intended task here to solve.
  
The creator of this challenge should have chosen a better name for the variable like `areAllAnimalsInEuropeWeighingLessThanOnehundred`. Then it would be clearer for us to solve this part of the challenge correctly.